public/app/percona/backup/components/BackupInventory/RestoreBackupModal/RestoreBackupModal.tsx

```diff
@@ -38,12 +42,111 @@ export const RestoreBackupModal: FC<RestoreBackupModalProps> = ({
 }) => {
   const styles = useStyles(getStyles);
   const initialValues = useMemo(() => (backup ? toFormProps(backup) : undefined), [backup]);
+
+  const [selectedTimerange, setSelectedTimerange] = useState<Timeranges>();
+  const [selectedTimerangeFromDatepicker, setSelectedTimerangeFromDatepicker] = useState<DateTime>();
+  const [selectedDay, setSelectedDay] = useState<Date | undefined>();
   const handleSubmit = ({ serviceType, service }: RestoreBackupFormProps) => {
-    if (backup) {
+    if (backup && selectedTimerangeFromDatepicker) {
       const serviceId = serviceType === ServiceTypeSelect.SAME ? backup.serviceId : service.value;
-      onRestore(serviceId || '', backup.id);
+      onRestore(serviceId || '', backup.id, selectedTimerangeFromDatepicker.toISOString());
     }
   };
+  const calculateDisableHours = useCallback(() => {
+    const disabledHours = [];
+
+    if (selectedTimerange) {
+      const { startTimestamp, endTimestamp } = selectedTimerange;
+      for (let i = 0; i < 24; i++) {
+        if (moment(selectedDay).isSame(startTimestamp, 'date')) {
```

Comment:
We should define this var out of the loop, it's always the same value.
Like `const isSameStartDay = moment(selectedDay).isSame(startTimestamp, 'date')`
Same for end.
Also, all moment constructors should be outside the loop

Reply:
Everything moved to utils